assets/js/vue/img.js

```diff
@@ -17,70 +27,110 @@ export default {
+
+        const dimensions = {
+          naturalHeight: height ? height : preset.fallback_max_width,
+          naturalWidth: width ? width : preset.fallback_max_width,
+        };
+        this.dimensions = dimensions;
+
+        const fallbackString = `${basePath}${transformationsString},w_${preset.fallback_max_width}/${this.img} ${this.dimensions.naturalWidth}w`;
+
+        height && width ? this.buildSrcSet(preset, transformationsString) : (this.fallback = fallbackString);
+      };
+
+      img.src = this.source;
     },
-    getPreset() {
-      const selectedPreset = Presets[this.preset];
+    getTransformationString(preset) {
+      const transformations = [];
+
+      for (const [key, value] of Object.entries(TransformationOptions)) {
+        if (preset[key]) {
+          transformations.push(`${value}_${preset[key]}`);
+        }
+      }
 
-      return selectedPreset ? selectedPreset : Presets['default'];
+      const transformationsIsNotEmpty = transformations.length > 0;
+
+      return transformationsIsNotEmpty ? transformations.join(',') : '';
     },
-    getPresetWidth(value) {
-      const width = value ? value : this.getPreset()?.fallback_max_width;
+    buildSrcSet(preset, transformationsString) {
+      const srcsetArray = [];
+      const steps = preset['steps'];
+      const minWidth = preset['min_width'];
+      const maxWidth = preset['max_width'];
+      const stepWidth = (maxWidth - minWidth) / (steps - 1);
+      const { naturalWidth } = this.dimensions;
+
+      for (let factor = 1; factor < steps; factor++) {
+        const width = minWidth + (factor - 1) * stepWidth;
+        const isWithinNaturalWidth = width <= naturalWidth;
+        const selectedWidth = isWithinNaturalWidth ? width : naturalWidth;
+        const srcsetString = `${basePath}${transformationsString},w_${selectedWidth}/${this.img} ${selectedWidth}w`;
 
-      if (!width) return '';
+        srcsetArray.push(srcsetString);
+
+        if (!isWithinNaturalWidth) break;
+      }
 
-      return `,w_${width}`;
+      this.srcset = naturalWidth < minWidth ? '' : srcsetArray.join(', \n');
     },
-    getUrlParams(value) {
-      if (this.isGif()) return '';
+    isGif() {
+      const extension = this.img.split('.')[1];
 
-      // TODO generate c_limit and stuff
-      return `c_limit,f_auto,q_auto,dpr_auto${this.getPresetWidth(value)}/`;
+      return extension.toLowerCase() === 'gif';
     },
   },
   props: {
     img: String,
+    alt: String,
     cloudinary: Boolean,
+    crossorigin: String,
     lazy: Boolean,
     class: String,
     preset: String,
   },
   template: `
-    <img :src="source" :loading="loading" :class="classList" :srcset="srcset" :sizes="sizes">
+    <img @load="loadImage" ref="image" :alt="this.alt" :src="source" :loading="loading" :class="classList" :width="this.dimensions.naturalWidth" :height="this.dimensions.naturalHeight" :srcset="this.srcset" :sizes="this.sizes" :crossorigin="this.crossorigin">
```

Comment:
Wenn man sich die Seite "cloud-security-operations-center" auf PROD und lokal ansieht, fällt mir noch auf, dass die "src" in Prod noch Parameter übergibt, was lokal nicht der Fall ist. zusätzlich wird eine weitere Größe im srcset generiert.